File: src/etch/examples/benchmark/etch/PerfListener.java
package etch.examples.benchmark.etch;

import metreos.core.io.Connection;
import metreos.core.io.Listener;
import metreos.core.io.ListenerHandler;
import metreos.util.Log;
import etch.bindings.java.support.Pool;

/**
 * Starts a Perf listener.
 */
public class PerfListener
{
	/**
	 * Program entry point.
	 * @param args
	 * @throws Exception
	 */
	public static void main( String[] args ) throws Exception
	{
		Log.addSink( null );
		Log.report( "PerfListener" );
		
		Pool queuedPool = null;
		Pool freePool = null;
		
		// create the listener
		
		ListenerHandler lh = PerfHelper.newPerfListener( queuedPool, freePool );
		Connection<ListenerHandler> l = new Listener( lh, 5, null, 4001, 0 );
		l.start();
	}
}

File: src/etch/examples/benchmark/etch/PerfListener.cs
using System;
using System.Collections.Generic;
using System.Text;

using Etch.Msg;
using Etch.Support;
using Etch.Transport;

namespace etch.examples.benchmark.etch
{
    public class PerfListener
    {

        public static void Main( String[] args )
        {
            Console.WriteLine( "PerfListener" );

            Pool queuedPool = null;
            Pool freePool = null;

            // create the listener
            ListenerHandler lh = PerfHelper.newPerfListener( queuedPool, freePool );
            Listener l = new Listener( lh, 5, "127.0.0.1", 4001, 0 );
            l.Start();
        }
    }
}
